node/podcontroller.go

```diff
@@ -245,11 +274,18 @@ func NewPodController(cfg PodControllerConfig) (*PodController, error) {
 		return nil, pkgerrors.Wrap(err, "could not create resource manager")
 	}
 
+	var provider PodUIDLifecycleHandler
```

Comment:
Please, add a comment explaining why. It will help us navigate this code in the future, especially when working to remove the non-UID Pod lifecycle handler.

Reply:
Also, related to this, I see the PodControllerConfig still require PodLifecycleHandler, so to my understanding (and also at my direct experience trying this), to use the UID version GetPod and GetPodStatus are still needed to be included although a dummy once. 

Is this the wanted behavior? We might need to document accordingly then.

Reply:
@dciangot This is addressed in the latest commit, there is a `UIDBasedProvider` function that can be used to wrap a `PodUIDLifecycleHandler` with the necessary stubs to be a `PodLifecycleHandler`. You can see it used in the mock provider (which no longer has `GetPod` and `GetPodStatus` methods)

In the future, we'd want to invert this and require a `PodUIDLifecycleHandler` and `PodNotifier` directly, but for now we will retain backwards compatibility.